src/journal/journal-file.c

```diff
@@ -1316,41 +1335,69 @@ int journal_file_find_field_object_with_hash(
 
                         if (ret)
                                 *ret = o;
-                        if (offset)
-                                *offset = p;
+                        if (ret_offset)
+                                *ret_offset = p;
 
                         return 1;
                 }
 
-                p = le64toh(o->field.next_hash_offset);
+                nextp = le64toh(READ_NOW(o->field.next_hash_offset));
+                if (nextp > 0) {
+                        if (nextp <= p) /* Refuse going in loops */
+                                return log_debug_errno(SYNTHETIC_ERRNO(EBADMSG),
+                                                       "Detected field hash item loop in %s, refusing.", f->path);
+
+                        depth++;
+
+                        /* If the depth of this hash chain is larger than all others we have seen so far, record it */
+                        if (JOURNAL_HEADER_CONTAINS(f->header, field_hash_chain_depth) && f->writable)
+                                f->header->field_hash_chain_depth = htole64(MAX(depth, le64toh(f->header->field_hash_chain_depth)));
```

Comment:
It look like this will unnecessarily write the same value into `f->header->field_hash_chain_depth` when it has not changed. Possibly this will be optimized out by the compiler, but it's still worrying.

Reply:
well, it's a structure in memory, it should not matter...